drivers/usb/udc/udc_mcux_ehci.c

```diff
@@ -525,10 +574,10 @@ usb_status_t USB_DeviceNotificationTrigger(void *handle, void *msg)
 	case kUSB_DeviceNotifyLPMSleep:
 		break;
 	case kUSB_DeviceNotifyDetach:
-		udc_submit_event(dev, UDC_EVT_VBUS_REMOVED, 0);
+		udc_mcux_change_state(dev, priv->enabled, 0);
```

Comment:
`udc_submit_event(dev, UDC_EVT_VBUS_REMOVED, 0);` should be submitted here regardless of other states.

Reply:
If we do the change I suggested in change_state() then this would occur in that function.

Reply:
I did the change mentioned by @dleach02 on line 109